black_it/loss_functions/likelihood.py

```diff
@@ -0,0 +1,127 @@
+# Black-box ABM Calibration Kit (Black-it)
+# Copyright (C) 2021-2022 Banca d'Italia
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""This module defines the 'LikelihoodLoss'  class."""
+import warnings
+from typing import Callable, List, Optional
+
+import numpy as np
+from numpy.typing import NDArray
+
+from black_it.loss_functions.base import BaseLoss
+
+TimeSeriesFilter = Optional[Callable[[NDArray[np.float64]], NDArray[np.float64]]]
+"""A filter that receives a time series and returns its filtered version. Used by the BaseLoss constructor."""
+
+
+def kernel(sq_dist: NDArray[np.float64], h: float, D: int) -> NDArray[np.float64]:
+    """Compute a kernel density estimation using a Gaussian kernel."""
+    print(h)
+    k = np.exp(-(sq_dist / (2 * h**2))) / (h**D * (2 * np.pi) ** (D / 2.0))
+    return k
+
+
+class LikelihoodLoss(BaseLoss):
+    """Class for a likelihood loss computation using a Gaussian kernel function."""
```

Comment:
we might want to add some references here if considered relevant (perhaps Grazzini et al 2017?)